types/nodepub/index.d.ts

```diff
@@ -0,0 +1,75 @@
+// Type definitions for nodepub 3.0
+// Project: https://github.com/kcartlidge/nodepub
+// Definitions by: Dylan Armstrong <https://github.com/dylanarmstrong>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface Links {
+  itemType: 'front' | 'contents' | 'main';
+  link: string;
+  title: string;
+}
+
+export type GenerateContentsCallback = (links?: Links) => string | void;
+
+export interface Metadata {
+  author: string;
+  cover: string;
+  id: number | string;
+  title: string;
+
+  contents?: string;
+  copyright?: string;
+  description?: string;
+  fileAs?: string;
+  genre?: string;
+  images?: string[];
+  language?: string;
+  published?: string;
+  publisher?: string;
+  sequence?: number;
+  series?: string;
+  showContents?: boolean;
+  source?: string;
+  tags?: string;
+
+  [key: string]: any;
```

Comment:
```suggestion
```
why do you think this objects takes whatever property? I cannot see a code that adds any property, just listed one

Reply:
Ah, I misread some stuff, it actually takes a few more than I realized though from the constituents folder.